src/dft/src/architect/ScanArchitectHeuristic.cpp

```diff
@@ -38,15 +38,84 @@
 #include "ClockDomain.hh"
 #include "ScanArchitect.hh"
 
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/index/rtree.hpp>
+
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
 namespace dft {
 
 ScanArchitectHeuristic::ScanArchitectHeuristic(
     const ScanArchitectConfig& config,
-    std::unique_ptr<ScanCellsBucket> scan_cells_bucket)
-    : ScanArchitect(config, std::move(scan_cells_bucket))
+    std::unique_ptr<ScanCellsBucket> scan_cells_bucket, utl::Logger* logger)
+    : ScanArchitect(config, std::move(scan_cells_bucket)), logger_(logger)
 {
 }
 
+void ScanArchitectHeuristic::order_reduce_wirelength(std::vector<std::unique_ptr<ScanCell>>& cells)
+{
+
+  // Nothing to order
+  if (cells.empty())
+    return;
+  // No point running this if the cells aren't placed yet
+  if (std::all_of(cells.begin(), cells.end(), [](const std::unique_ptr<ScanCell> &cell) {
+        return !cell->isPlaced();
+      }))
+    return;
```

Comment:
I do generally like std::all_of, but in this case I think 

```c++
for (const auto& cell: cells) {
  if (!cell->isPlaced()) {
     return;
  }
}
```

would be a little bit clearer.